course_flow/models.py

```diff
@@ -7,23 +7,54 @@
 from django.db.models.signals import pre_delete, post_save
 from django.dispatch import receiver
 from django.utils.translation import ugettext_lazy as _
+from model_utils.managers import InheritanceManager
 
 
 User = get_user_model()
 
 
 class Column(models.Model):
-    title = models.CharField(max_length=50)
+    title = models.CharField(max_length=50,null=True, blank=True)
     author = models.ForeignKey(User, on_delete=models.SET_NULL, null=True)
     created_on = models.DateTimeField(auto_now_add=True)
     last_modified = models.DateTimeField(auto_now=True)
+    CUS_ACT = 0
+    OOCI=1
+    OOCS=2
+    ICI=3
+    ICS=4
+    CUS_CO=10
+    PREP=11
+    LESSON=12
+    ARTIFACT=13
+    ASSESS=14
+    CUS_PROG=20
     
-    default_activity_columns = ["ooci", "ooc", "ici", "ics"]
+    COLUMN_TYPES = (
+        (CUS_ACT,"Custom Activity Column"),
+        (OOCI,"Out of Class (Instructor)"),
+        (OOCS,"Out of Class (Students)"),
+        (ICI,"In Class (Instructor)"),
+        (ICS,"In Class (Students)"),
+        (CUS_CO,"Custom Course Column"),
+        (PREP,"Preparation"),
+        (LESSON,"Lesson"),
+        (ARTIFACT,"Artifact"),
+        (ASSESS,"Assessment"),
+        (CUS_PROG,"Custom Program Category"),
+    )
+    column_type = models.PositiveIntegerField(default=0,choices=COLUMN_TYPES)    
+    
+    default_columns = {
+        'activity':[1,2,3,4],
+        'course':[11,12,13,14],
+        'program':[20,20,20],
+    }
 
     hash = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
 
     def __str__(self):
-        return self.title
+        return str(self.column_type)
```

Comment:
Yeah this was actually intentional, I just wanted the integer. But I realize now that this can lead to breakage since it needs to return a string. I'm fixing these.